packages/blockchain-utils-linking/src/__tests__/solana.test.ts

```diff
@@ -0,0 +1,54 @@
+import { MessageSignerWalletAdapterProps } from '@solana/wallet-adapter-base';
+import { SOLANA_MAINNET_CHAIN_REF, SolanaAuthProvider } from '../solana'
+import { Keypair } from '@solana/web3.js';
+import { Buffer } from 'buffer';
+import * as nacl from 'tweetnacl';
+import * as uint8arrays from 'uint8arrays'
+
+const did = 'did:3:bafysdfwefwe'
+const privKey = 'mdqVWeFekT7pqy5T49+tV12jO0m+ESW7ki4zSU9JiCgbL0kJbj5dvQ/PqcDAzZLZqzshVEs01d1KZdmLh4uZIg=='
+const chainRef = SOLANA_MAINNET_CHAIN_REF
+
+
+class MyWalletAdapter implements MessageSignerWalletAdapterProps {
+  readonly _keyPair: Keypair
+
+  constructor(keyPair: Keypair) {
+    this._keyPair = keyPair
+  }
+
+  async signMessage(message: Uint8Array): Promise<Uint8Array> {
+    return nacl.sign.detached(message, this._keyPair.secretKey)
```

Comment:
Please use `@stablelib/ed25519` instead of `tweetnacl` (same author).

Reply:
for sure!  will change it